### PR TITLE
Added checking the log level

### PR DESCRIPTION
**Pull request by gburanov**
We need to add the function to check the current log type.
That will help us to write uglier, but more efficient code

Some measurements from this PR:
https://github.com/remerge/go-tracker/pull/8

When changing from
```
		log.WithFields(cue.Fields{
			"topic":   topic,
			"message": string(buf),
		}).Debug("sending safe message")
```
to
```
	if log.EnabledFor(cue.DEBUG) {
		log.WithFields(cue.Fields{
			"topic":   topic,
			"message": string(buf),
		}).Debug("sending safe message")
	}
```

Reduces the number of heap allocations from 18489 to 14302

### CODE REVIEW

File: logger.go [hunk 1]
```diff
@@ -231,6 +232,14 @@ func (l *logger) ReportRecovery(cause interface{}, message string) {
 	l.sendRecovery(cause, message)
 }
 
+func (l *logger) EnabledFor(level Level) bool {
+	config := cfg.get()
+	if level > config.threshold {
```
Comment (akaspin) [inline logger.go]: Maybe just `return level < config.threshold`?

**Review by akaspin — CHANGES_REQUESTED** [review]

File: logger.go (hunk 1 above)
Comment (gburanov) [reply logger.go]: yes, sure)

Comment (gburanov) [reply logger.go]: actually level <= config.threshold

**Review by akaspin — APPROVED** [review]
👍

**Comment by gburanov** [comment]
Alternative implementation https://github.com/remerge/cue/pull/3

**Review by dynamix — APPROVED** [review]
Looks good. I like this more than the `SmartSend` in #3 because it still allows you to use `.WithValue` which doesn't create a cue.Fields if you only have a single k/v pair and formatting of the string with varargs.

Having said that. Why not just a method that returns the log level?

**Comment by gburanov** [comment]
@dynamix  for me this method will be confusing. For example if it will return Debug, than this does not actually mean that INFO messages are not logged

**Pull request merged**